pkg/executor/join.go

```diff
@@ -332,7 +332,7 @@ func (w *buildWorker) fetchBuildSideRows(ctx context.Context, chkCh chan<- *chun
 		if w.hashJoinCtx.finished.Load() {
 			return
 		}
-		chk := sessVars.GetNewChunkWithCapacity(w.buildSideExec.RetFieldTypes(), sessVars.MaxChunkSize, sessVars.MaxChunkSize, w.hashJoinCtx.allocPool)
+		chk := w.hashJoinCtx.allocPool.Alloc(w.buildSideExec.RetFieldTypes(), sessVars.MaxChunkSize, sessVars.MaxChunkSize)
```

Comment:
By the way, the use of `MaxChunkSize` is not reasonable, we met Go GC issues when the column length is varchar(1500)/varchar(3000)... in that case the chunk object size amplify too much and cause high pressue on Go GC.

Reply:
We can take a close look of all `sessVars.MaxChunkSize` chunk allocating later.

Reply:
Do you think it would be better to also use `BaseExecutor.InitCap` and `BaseExecutor.MaxChunkSize` here? Though the current context doesn't have this information, we can pass them (to the `hashJoinCtx`).

Reply:
I haven't check the code for how `MaxChunkSize` is used yet, so no suggestions.